src/EPR.Payment.Service.Data.UnitTests/Repositories/Payments/PaymentsRepositoryTests.cs

```diff
@@ -0,0 +1,45 @@
+﻿using AutoFixture.MSTest;
+using EPR.Payment.Service.Common.Data.Interfaces;
+using EPR.Payment.Service.Common.Data.Repositories.Payments;
+using EPR.Payment.Service.Common.UnitTests.Mocks;
+using EPR.Payment.Service.Common.UnitTests.TestHelpers;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Moq.EntityFrameworkCore;
+using System.Data.Entity;
+
+namespace EPR.Payment.Service.Data.UnitTests.Repositories.Payments
+{
+    [TestClass]
+    public class PaymentsRepositoryTests
+    {
+        private Mock<DbSet<Common.Data.DataModels.Payment>> _paymentMock = null!;
+        private CancellationToken _cancellationToken;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            _paymentMock = MockIPaymentRepository.GetPaymentMock();
+            _cancellationToken = new CancellationToken();
+        }
+
+        [TestMethod, AutoMoqData]
+        public async Task GetPreviousPaymentsByReferenceAsync_PaymentsExist_ShouldReturnPreviousAmounts(
+            [Frozen] Mock<IAppDbContext> _dataContextMock,
+            [Greedy] PaymentsRepository _mockPaymentsRepository)
+        {
+            //Arrange
+            _dataContextMock.Setup(i => i.Payment).ReturnsDbSet(_paymentMock.Object);
+            _mockPaymentsRepository = new PaymentsRepository(_dataContextMock.Object);
+
+            var reference = "Test 1 Reference";
+
+            //Act
+            var result = await _mockPaymentsRepository.GetPreviousPaymentsByReferenceAsync(reference, _cancellationToken);
+
+            //Assert
+            result.Should().Be(10.0m);
+        }
+    }
```

Comment:
I think it would be worth adding an additional test here to pick up the scenario where multiple payments exist in the database.

Reply:
done